src/zenml/model/model_config.py

```diff
@@ -123,40 +296,42 @@ def _create_model_version(
                 model_name_or_id=self.name,
                 model_version_name_or_number_or_id=self.version,
             )
-            self._model_version = mv
+            self.model_version = mv
         except KeyError:
-            self._model_version = zenml_client.create_model_version(
+            self.model_version = zenml_client.create_model_version(
                 model_version=mv_request
             )
             logger.info(f"New model version `{self.version}` was created.")
 
-        return self._model_version
+        return self.model_version
 
     def _get_model_version(self) -> "ModelVersionResponseModel":
         """This method gets a model version from Model Control Plane.
 
         Returns:
             The model version based on configuration.
         """
-        if self._model_version is not None:
-            return self._model_version
+        if self.model_version is not None:
+            from zenml.models.model_models import ModelVersionResponseModel
+
+            return ModelVersionResponseModel(**dict(self.model_version))
```

Comment:
Isn't `self.model_version` already a `ModelVersionResponseModel`?

Reply:
I'll redesign this

Reply:
I decided to get rid of this caching, as it produces unexpected results and always returns the version cached at the very beginning of the pipeline run.